### PR TITLE
Add new command "export_transfers" to save transfers to csv

### PR DESCRIPTION
**Pull request by sachaaaaa**
Since `show_transfers` and `export_transfers` basically require the same data, I've added a new function `get_transfers` that returns a vector of `transfer_view` structs (new structure as well). That function does all the querying, calculations and sorting.
Then, it's up to the caller function to display the data in the appropriate way (print or export as csv).

Notable difference between the csv file and the printed list of transfers:
- The csv file shows the "running balance" next the the transfer amount. This field is not affected by unconfirmed transfers (pending/failed/pool).
- To handle transfers with multiple outputs, the csv file displays a new line for any subsequent outputs after the first one,  showing only the destination address and amount sent:
```
x, <total amount>, xxxxx, xxxxxxxxxxxxxxxxx, <address1>, <amount 1>
 ,               ,      ,                  , <address2>, <amount 2>
```

Changes made to both:
- display the (shortened) destination address for incoming transfers, to easily match which address the funds were sent to. This will make it easier to match the transfer with the owner's accounts/addresses.

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -6327,20 +6337,183 @@ bool simple_wallet::show_transfers(const std::vector<std::string> &args_)
...
+      }
+    }
+  }
+  // sort by block, then by timestamp (unconfirmed last)
+  std::sort(transfers.begin(), transfers.end(), [](const transfer_view& a, const transfer_view& b) -> bool {
+    if (a.confirmed && !b.confirmed)
+      return true;
+    if (a.block == b.block)
+      return a.timestamp < b.timestamp;
+    return a.block < b.block;
```
Comment (jcktm) [inline src/simplewallet/simplewallet.cpp]: Sort order for unconfirmed transactions may differ depending on the translation, not sure if you wanted a strict ordering for failed/pending/pool... probably doesn't matter

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -6251,50 +6244,54 @@ bool simple_wallet::show_transfers(const std::vector<std::string> &args_)
...
-          dests += ", ";
-        dests +=  get_account_address_as_str(m_wallet->nettype(), d.is_subaddress, d.addr) + ": " + print_money(d.amount);
+        destinations.push_back({get_account_address_as_str(m_wallet->nettype(), d.is_subaddress, d.addr), d.amount});
       }
       std::string payment_id = string_tools::pod_to_hex(i->second.m_payment_id);
       if (payment_id.substr(16).find_first_not_of('0') == std::string::npos)
         payment_id = payment_id.substr(0,16);
       std::string note = m_wallet->get_tx_note(i->first);
-      output.insert(std::make_pair(pd.m_block_height, std::make_pair(false, (boost::format("%16.16s %20.20s %s %s %14.14s %s %s - %s") % get_human_readable_timestamp(pd.m_timestamp) % print_money(pd.m_amount_in - change - fee) % string_tools::pod_to_hex(i->first) % payment_id % print_money(fee) % dests % print_subaddr_indices(pd.m_subaddr_indices) % note).str())));
+      transfers.push_back({
```
Comment (Doy-lee) [inline src/simplewallet/simplewallet.cpp]: Waiit. I think this pushes up the minimum to C++17/20. I talked to Sacha about this before and I think he was intending to change this to not use designated initialisers.

Comment (jcktm) [reply src/simplewallet/simplewallet.cpp]: Oh, I merged it.. my bad. Let's just fix in another commit

**Pull request merged**